doc/HLD_Version_Numbers.rst

```diff
@@ -0,0 +1,231 @@
+
+Conformance
+===============
+
+- [r.verno.serial]: version number ordering is compatible with ordering of events in the unit serial history. Indeed, version numbers are ordered by lsn-s or, equivalently (thanks to the version number ordering invariant), by version counters and the latter are (by the version number update protocol) in the same order as updates in the serial history; 
+
+- [r.verno.resource]: a client (or a proxy server) obtains the current value of a unit's version counter together with the update lock on the unit. The client can then increment this counter multiple times to produce a stream of local version numbers. When corresponding state updates are re-integrated to the server, matching version numbers are produced by using server fol lsn-s. This way, the clients can independently generate version numbers usable on a server; 
+
+- [r.verno.dtm]: this is a consequence of [r.verno.serial] fulfillment: if updates are replayed in the order of their version numbers, original serial history is exactly reproduced; 
+
+- [r.verno.optimistic-concurrency]: version numbers can be used for scalable generation of pseudo-times are discussed in [1]; 
+
+- [r.verno.update-streams]: as described in the cache re-integration section of the Functional specification, update stream mechanism can be implementing by introducing a special unit tracking re-integration state. This unit servers as a transactional stream caret, tracking re-integration progress in the presence of failures; 
+
+- [r.verno.fol]: by design, version number contains lsn field.
+
+Dependencies
+===============
+
+- [r.dtx]: distributed transactions are supported 
+
+  - [r.dtx.units]: file system updates acts on units 
+
+  - [r.dtx.units.identity]: units affected by a given update are identifiable 
+
+  - [R.DTX.SERIALIZABLE]: updates are serializable 
+
+- [r.resource]: scalable distributed resource management is implemented 
+
+  - [r.resource.cacheable]: resources are cacheable by consumers 
+
+- rpc: 
+
+  - [R.NET.RPC.STREAMS.MULTIPLE]: multiple update concurrent streams between a given pair of nodes are supported 
+
+- fol 
+
+  - [r.fol.lsn]: fol records are identifiable by lsn 
+
+  - [r.fol.verno]: fol record stores before- and after- version numbers 
+
+  - [r.fol.lsn.compare]: lsn numbers can be compared 
+
+- back-end 
+
+  - [r.back-end.store-verno]: back end stores current version number as object attribute 
+
+  - [R.BACK-END.TRANSACTIONAL]: back-end supports local transactions
+  
+Security Model
+===============
+
+A node can wreak chaos in the system by spoofing version numbers, so they must be protected by the same means as the rest of meta-data.
+
+Refinement
+===============
+
+Version number component has little internal logic. 
+
+***************
+State
+***************
+
+See version number comparison invariant above. 
+
+Use Cases (Scenarios)
+=====================
+
+Scenario 1
+
+.. image:: Images/scev1.PNG
```

Comment:
Remove all the IMAGES here, and replace them with tables.

Reply:
Incorporated.